### PR TITLE
Support remote development

### LINKED ISSUE #129: Remote WSL: "File: duplicate" command does not work
**Describe the bug**
Using the "File: duplicate" command does not work. The following error is logged to dev tools:

```
ERR cb.apply is not a function: TypeError: cb.apply is not a function
	at /home/…/.vscode-server/bin/2213894ea0415ee8c85c5eea0d0ff81ecc191529/node_modules/graceful-fs/polyfills.js:287:18
	at FSReqWrap.oncomplete (fs.js:155:5)
```

**To Reproduce**
1. Open a file in VSCode WSL
2. Attempt to duplicate a file (via explorer context menu OR command palette)
3. Nothing happens, error in console.

**Expected behavior**
The file should be duplicated.

**Desktop (please complete the following information):**
- VSCode Version: 1.36.1
- OS Version: Windows 10 Pro 10.0.17763 Build 17763
- WSL Linux Distro: Ubuntu 18.04

### PR DESCRIPTION
**Pull request by sleistner**
# Description

This PR adds support to manipulate files and folders in remote environments such as Docker containers or WSL.

Fixes #129

### CODE REVIEW

File: test/helper/stubs.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+export function stubGetConfiguration(returns?: any): sinon.SinonStub {
+    return stubObject(workspace, 'getConfiguration', returns);
+}
+
+export function restoreGetConfiguration() {
+    restoreObject(workspace.getConfiguration);
+}
+
+export function stubInputBox(returns?: any): sinon.SinonStub {
```
Comment (alexsavio) [inline test/helper/stubs.ts]: where is the verb in these function names?

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// See https://aka.ms/vscode-remote/devcontainer.json for format details.
+{
+	"name": "Node.js 8 & TypeScript",
+	"dockerFile": "Dockerfile",
+	// Uncomment the next line if you want to publish any ports.
```
[the remark this answers is not in the document]
Comment (sleistner) [reply .devcontainer/devcontainer.json]: done

File: test/helper/stubs.ts (hunk 1 above)
Comment (sleistner) [reply test/helper/stubs.ts]: verb added

**Pull request merged**